axelor-sale/src/main/java/com/axelor/apps/sale/service/saleorder/SaleOrderInitValueServiceImpl.java

```diff
@@ -0,0 +1,161 @@
+    saleOrderMap.put("advanceTotal", BigDecimal.ZERO);
+
+    saleOrderMap.put("creationDate", appBaseService.getTodayDate(saleOrder.getCompany()));
+    saleOrderMap.put("statusSelect", SaleOrderRepository.STATUS_DRAFT_QUOTATION);
+    saleOrderMap.put("company", company);
+
+    saleOrderMap.put("salespersonUser", getUser(saleOrder, salesPersonSelect));
+    saleOrderMap.put("team", userService.getUserActiveTeam());
+    saleOrderMap.put("printingSettings", printingSettings);
+    if (user != null) {
+      saleOrderMap.put("tradingName", user.getTradingName());
+    }
+    saleOrderMap.put("template", false);
+
+    if (company != null) {
+      saleOrderMap.put("duration", company.getSaleConfig().getDefaultValidityDuration());
+      saleOrderMap.put("currency", company.getCurrency());
+    }
+
+    for (Map.Entry<String, Object> entry : saleOrderMap.entrySet()) {
+      Mapper.of(SaleOrder.class).set(saleOrder, entry.getKey(), entry.getValue());
+    }
+
+    return saleOrderMap;
+  }
+
+  protected Map<String, Object> getInAti(SaleOrder saleOrder) {
+    Map<String, Object> saleOrderMap = new HashMap<>();
+    Company company = getCompany();
+    if (company != null) {
+      int saleOrderInAtiSelect = company.getSaleConfig().getSaleOrderInAtiSelect();
+      boolean inAti = saleOrderInAtiSelect == 2 || saleOrderInAtiSelect == 4;
+      saleOrder.setInAti(inAti);
+    }
+    saleOrderMap.put("inAti", saleOrder.getInAti());
+    return saleOrderMap;
+  }
+
+  protected Map<String, Object> getBankDetails(SaleOrder saleOrder) throws AxelorException {
+    Map<String, Object> saleOrderMap = new HashMap<>();
+    saleOrder.setCompanyBankDetails(
+        bankDetailsService.getDefaultCompanyBankDetails(
+            saleOrder.getCompany(), null, saleOrder.getClientPartner(), null));
+    saleOrderMap.put("companyBankDetails", saleOrder.getCompanyBankDetails());
+    return saleOrderMap;
+  }
+
+  protected Map<String, Object> getGroupProductsOnPrintings(SaleOrder saleOrder)
+      throws AxelorException {
+    Map<String, Object> saleOrderMap = new HashMap<>();
+    AppBase appBase = appBaseService.getAppBase();
+    saleOrder.setGroupProductsOnPrintings(appBase.getIsRegroupProductsOnPrintings());
+    saleOrderMap.put("companyBankDetails", saleOrder.getGroupProductsOnPrintings());
+    return saleOrderMap;
+  }
+
+  protected User getUser(SaleOrder saleOrder, int salesPersonSelect) {
+    User user = null;
+    Partner clientPartner = saleOrder.getClientPartner();
+    if (salesPersonSelect == 1) {
+      user = AuthUtils.getUser();
+    }
+    if (salesPersonSelect == 2 && clientPartner != null) {
+      user = clientPartner.getUser();
+    }
+    return user;
+  }
+
+  protected Company getCompany() {
+    Company activeCompany = AuthUtils.getUser().getActiveCompany();
+    if (activeCompany == null && companyRepository.all().count() == 1) {
+      activeCompany = companyRepository.all().fetchOne();
+    }
+    return activeCompany;
+  }
```

Comment:
I think this method could be used at a lot of places. Please move it into a separated service